### PR TITLE
[OSDEV-923] Add more logs around API and list uploads

### PR DESCRIPTION
**Pull request by roman-stolar**
[OSDEV-923](https://opensupplyhub.atlassian.net/browse/OSDEV-923) Add more logs around API and list uploads
- Added additional logs with steps & details data to API upload
- Added additional logs with steps & details data to List upload
- Added additional logs with steps & details data to Dedupe Hub match processing

### CODE REVIEW

**Comment by vladsha-dev** [comment]
@roman-stolar please update the RELEASE-NOTES.md file

File: doc/release/RELEASE-NOTES.md [hunk 1]
```diff
@@ -3,6 +3,26 @@ All notable changes to this project will be documented in this file.
 
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html). The format is based on the `RELEASE-NOTES-TEMPLATE.md` file.
 
+## Release 1.11.0
+
+## Introduction
+* Product name: Open Supply Hub
+* Release date: April 15, 2024
```
Comment (mazursasha1990) [inline doc/release/RELEASE-NOTES.md:10]: Please, change the release date to April 20, 2024

File: src/dedupe-hub/api/app/matching/matcher/exact/exact_matcher.py [hunk 2]
```diff
@@ -24,12 +31,18 @@ def process(
     ) -> Dict[str, List[MatchDTO]]:
         self.set_start()
 
+        log.info(f'[Matching] Exact match processing started!')
```
Comment (mazursasha1990) [inline src/dedupe-hub/api/app/matching/matcher/exact/exact_matcher.py:34]: Please, remove unnecessary `f-string`

File: src/dedupe-hub/api/app/matching/matcher/gazeteer/gazetteer_matcher.py [hunk 3]
```diff
@@ -46,6 +53,10 @@ def __init__(
 
     def process(self, messy: FacilityListItemDict) -> Dict[str, List[MatchDTO]]:
         self.set_start()
+
+        log.info(f'[Matching] Gazetteer match processing started!')
```
Comment (mazursasha1990) [inline src/dedupe-hub/api/app/matching/matcher/gazeteer/gazetteer_matcher.py:57]: Please, remove unnecessary f-string

File: src/dedupe-hub/api/app/matching/matcher/gazeteer/gazetteer_matcher.py [hunk 4]
```diff
@@ -72,9 +83,11 @@ def gazetter_match(self, messy):
                 generator=True
             )
         except NoCanonicalRecordsError:
+            log.error('[Matching] Error: No canonical records')
```
Comment (mazursasha1990) [inline src/dedupe-hub/api/app/matching/matcher/gazeteer/gazetteer_matcher.py:86]: Please, remove unnecessary f-string

File: src/dedupe-hub/api/app/matching/matcher/gazeteer/gazetteer_matcher.py [hunk 5]
```diff
@@ -99,6 +112,9 @@ def filter_matches(
                         'score': float(score)
                         })
 
+        log.info(f'[Matching] Gazetteer match processing finished!')
```
Comment (mazursasha1990) [inline src/dedupe-hub/api/app/matching/matcher/gazeteer/gazetteer_matcher.py:115]: Please, remove unnecessary `f-string`

File: src/dedupe-hub/api/app/matching/matcher/gazeteer/gazetteer_matcher.py (hunk 4 above)
Comment (roman-stolar) [reply src/dedupe-hub/api/app/matching/matcher/gazeteer/gazetteer_matcher.py:86]: There is no f-string

**Review by mazursasha1990 — APPROVED** [review]
LGTM

File: src/dedupe-hub/api/app/matching/matcher/exact/exact_matcher.py [hunk 6]
```diff
@@ -24,12 +31,18 @@ def process(
     ) -> Dict[str, List[MatchDTO]]:
         self.set_start()
 
+        log.info('[Matching] Exact match processing started!')
+        log.info(f'[Matching] Messy data: {messy}')
```
Comment (killev) [inline src/dedupe-hub/api/app/matching/matcher/exact/exact_matcher.py:35]: do you think it's good to print all messy data?

Comment (roman-stolar) [reply src/dedupe-hub/api/app/matching/matcher/exact/exact_matcher.py:35]: @killev messy is "data that user upload" here.

Comment (roman-stolar) [reply src/dedupe-hub/api/app/matching/matcher/exact/exact_matcher.py:35]: So I think to see what come into Exact Matcher and then see what coming out from there. The same for Gazzetteer Matcher. @killev

**Review by killev — APPROVED** [review]
LGTM

**Pull request merged**